### PR TITLE
Refactor pagination filters for custom assets

### PR DESCRIPTION
**Pull request by tewshi**
## Checklist

- [x] The PR modified the frontend, and updated the custom assets component to use pagination/filter composable

### CODE REVIEW

File: frontend/app/src/types/asset/index.ts [hunk 1]
```diff
@@ -102,7 +103,8 @@ export const CustomAssets = CollectionCommonFields.extend({
 });
 export type CustomAssets = z.infer<typeof CustomAssets>;
 
-export interface CustomAssetPagination extends ApiPagination<CustomAsset> {
+export interface CustomAssetRequestPayload
+  extends PaginationRequestPayload<CustomAsset> {
```
Comment (kelsos) [inline frontend/app/src/types/asset/index.ts:107]: can you check if ApiPagination/PaginationRequestPayload are overlapping?

**Review by kelsos — APPROVED** [review]
lgtm, I am approving and merging this we can check and maybe address the comments in the next PR :)

File: frontend/app/src/components/asset-manager/CustomAssetTable.vue [hunk 2]
```diff
@@ -4,37 +4,31 @@ import { type PropType, type Ref } from 'vue';
...
 
-const props = defineProps({
+defineProps({
   assets: { required: true, type: Array as PropType<CustomAsset[]> },
   loading: { required: false, type: Boolean, default: false },
   serverItemLength: { required: true, type: Number },
-  types: { required: true, type: Array as PropType<string[]> }
+  matchers: { required: true, type: Array as PropType<Matcher[]> },
+  filters: { required: true, type: Object as PropType<Filters> }
 });
```
[the remark this answers is not in the document]
Comment (tewshi) [reply frontend/app/src/components/asset-manager/CustomAssetTable.vue:20]: Okay, will update it, thanks.

[the remark this answers is not in the document]
Comment (tewshi) [reply frontend/app/src/components/asset-manager/CustomAssetTable.vue:20]: done

File: frontend/app/src/types/asset/index.ts (hunk 1 above)
Comment (tewshi) [reply frontend/app/src/types/asset/index.ts:107]: checking

Comment (tewshi) [reply frontend/app/src/types/asset/index.ts:107]: yes, they do, but these 2 are extras in `PaginationRequestPayload`

```typescript
readonly ignoreCache?: boolean;
readonly onlyCache?: boolean;
```

Comment (tewshi) [reply frontend/app/src/types/asset/index.ts:107]: and these are not required in `PaginationRequestPayload`

```typescriopt
readonly orderByAttributes?: (keyof T)[];
readonly ascending?: boolean[];
```

**Pull request merged**